internal/namespaces/instance/v1/custom_server_ssh.go

```diff
@@ -0,0 +1,85 @@
+package instance
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"reflect"
+
+	"github.com/scaleway/scaleway-cli/internal/core"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+type instanceSSHServerRequest struct {
+	Zone     scw.Zone
+	ServerID string
+	Username string
+	Port     uint
+}
+
+func serverSSHCommand() *core.Command {
+	return &core.Command{
+		Short:     `SSH into a server`,
+		Long:      `Connect to distant server via the SSH protocol.`,
+		Namespace: "instance",
+		Verb:      "ssh",
+		Resource:  "server",
+		ArgsType:  reflect.TypeOf(instanceSSHServerRequest{}),
+		ArgSpecs: core.ArgSpecs{
+			{
+				Name:       "server-id",
+				Short:      "Server ID to SSH into",
+				Required:   true,
+				Positional: true,
+			},
+			{
+				Name:    "username",
+				Short:   "Username used for the SSH connection",
+				Default: core.DefaultValueSetter("root"),
+			},
+			{
+				Name:    "port",
+				Short:   "Port used for the SSH connection",
+				Default: core.DefaultValueSetter("22"),
+			},
+			core.ZoneArgSpec(),
+		},
+		Run: instanceServerSSHRun,
+	}
+}
+
+func instanceServerSSHRun(ctx context.Context, argsI interface{}) (i interface{}, e error) {
+	args := argsI.(*instanceSSHServerRequest)
+
+	client := core.ExtractClient(ctx)
+	apiInstance := instance.NewAPI(client)
+	serverResp, err := apiInstance.GetServer(&instance.GetServerRequest{
+		Zone:     args.Zone,
+		ServerID: args.ServerID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if serverResp.Server.State != instance.ServerStateRunning {
+		return nil, fmt.Errorf("server is not running")
+	}
+
+	if serverResp.Server.PublicIP == nil {
+		return nil, fmt.Errorf("server does not have a public IP to connect to")
```

Comment:
Same here add a hint with a command to add an IP
IIRC `scw instance server update ip=new`